src/components/templates/PartyMap/components/RoomModal/RoomModal.tsx

```diff
@@ -24,53 +21,21 @@ import { RoomModalOngoingEvent, ScheduleItem } from "..";
 import "./RoomModal.scss";
 
 interface RoomModalProps {
-  show: boolean;
   onHide: () => void;
-  room?: Room;
+  room: Room;
 }
 
-export const RoomModal: React.FC<RoomModalProps> = ({ show, onHide, room }) => {
-  const { user, profile } = useUser();
-
-  const venues = useSelector(orderedVenuesSelector);
+export const RoomModal: React.FC<RoomModalProps> = ({ onHide, room }) => {
   const venueEvents = useSelector(venueEventsSelector) ?? [];
 
-  const roomTitle = room?.title;
-  const userLastSeenIn = profile?.lastSeenIn;
-
-  const { recentRoomUsers } = useRecentRoomUsers(roomTitle);
-
-  const roomVenue = useMemo(() => {
-    if (!room) return undefined;
-
-    return venues?.find((venue) => room.url.endsWith(`/${venue.id}`));
-  }, [room, venues]);
-
-  const venueName = roomVenue?.name;
-
-  // TODO: @debt refactor this to use openRoomWithCounting
-  const enter = useCallback(() => {
-    if (!room || !user) return;
+  const { enterRoom, recentRoomUsers } = useRoom(room);
 
-    const nowInMilliseconds = getCurrentTimeInMilliseconds();
-
-    const venueRoom = venueName ? { [venueName]: nowInMilliseconds } : {};
-
-    trackRoomEntered(
-      user,
-      {
-        [`${venueName}/${roomTitle}`]: nowInMilliseconds,
-        ...venueRoom,
-      },
-      userLastSeenIn
-    );
-
-    if (roomVenue) {
-      openUrl(venueInsideUrl(roomVenue.id));
-      return;
-    }
-    openRoomUrl(room.url);
-  }, [userLastSeenIn, room, roomTitle, user, venueName, roomVenue]);
+  useMousetrap({
```

Comment:
Moved from useKeyboard hook

Reply:
Why was this moved from the `useKeyboard hook` to here?